### PR TITLE
Ref Token changes plus JFrog repo order changes

### PR DESCRIPTION
**Pull request by sumanthkb44**
Ref Token changes plus JFrog repo order changes

### CODE REVIEW

File: TestFiles/IntegrationTestFiles/ArtifactoryUploaderTestData/02-22-2024_1031_Backup_Test_Bom.cdx.json
Comment (adityanarayanp) [inline TestFiles/IntegrationTestFiles/ArtifactoryUploaderTestData/02-22-2024_1031_Backup_Test_Bom.cdx.json:1]: Is this file required ?

File: TestFiles/IntegrationTestFiles/ArtifactoryUploaderTestData/02-22-2024_1032_Backup_Test_Bom.cdx.json
Comment (adityanarayanp) [inline TestFiles/IntegrationTestFiles/ArtifactoryUploaderTestData/02-22-2024_1032_Backup_Test_Bom.cdx.json:1]: Is this file required ?

File: TestFiles/IntegrationTestFiles/ArtifactoryUploaderTestData/Test_Bom.cdx.json
Comment (adityanarayanp) [inline TestFiles/IntegrationTestFiles/ArtifactoryUploaderTestData/Test_Bom.cdx.json:1]: Is this file required ?

File: src/LCT.APICommunications/Interfaces/IJfrogAqlApiCommunication.cs [hunk 1]
```diff
@@ -29,5 +29,7 @@ public interface IJfrogAqlApiCommunication
         /// <param name="path">repoName</param>
         /// <returns>AqlResult</returns>
         Task<HttpResponseMessage> GetPackageInfo(string repoName, string packageName = null, string path = null);
+
```
Comment (adityanarayanp) [inline src/LCT.APICommunications/Interfaces/IJfrogAqlApiCommunication.cs:32]: Please add method description in the similar lines of the GetPackageInfo method

File: src/LCT.APICommunications/JfrogAqlApiCommunication.cs [hunk 2]
```diff
@@ -80,9 +88,8 @@ public async Task<HttpResponseMessage> GetPackageInfo(string repoName, string pa
         private static HttpClient GetHttpClient(ArtifactoryCredentials credentials)
         {
             HttpClient httpClient = new HttpClient();
-
-            httpClient.DefaultRequestHeaders.Add(ApiConstant.JFrog_API_Header, credentials.ApiKey);
-            httpClient.DefaultRequestHeaders.Add(ApiConstant.Email, credentials.Email);
```
Comment (adityanarayanp) [inline src/LCT.APICommunications/JfrogAqlApiCommunication.cs:85]: If email is no longer required, we need to also remove it from the app settings also

File: src/LCT.Facade/JfrogAqlApiCommunicationFacade.cs [hunk 3]
```diff
@@ -27,6 +27,11 @@ public JfrogAqlApiCommunicationFacade(IJfrogAqlApiCommunication jfrogAqlApiCommu
             m_jfrogAqlApiCommunication = jfrogAqlApiCommunication;
         }
 
+        public Task<HttpResponseMessage> CheckConnection()
```
Comment (adityanarayanp) [inline src/LCT.Facade/JfrogAqlApiCommunicationFacade.cs:30]: Please add summary

File: src/LCT.Facade/Interfaces/IJfrogAqlApiCommunicationFacade.cs [hunk 4]
```diff
@@ -29,5 +29,7 @@ public interface IJfrogAqlApiCommunicationFacade
         /// <param name="path">repoName</param>
         /// <returns>AqlResult</returns>
         Task<HttpResponseMessage> GetPackageInfo(string repoName, string packageName, string path);
+
```
Comment (adityanarayanp) [inline src/LCT.Facade/Interfaces/IJfrogAqlApiCommunicationFacade.cs:32]: Please add summary

File: src/LCT.PackageIdentifier/BomCreator.cs [hunk 5]
```diff
@@ -165,5 +166,27 @@ private async Task<Bom> ComponentIdentification(CommonAppSettings appSettings, I
...
         }
+
+        public async Task<bool> CheckJFrogConnection()
+        {
+            var response = await JFrogService.CheckJFrogConnectivity();
+            if (response != null)
+            {
+                if (response.IsSuccessStatusCode)
+                {
+                    Logger.Logger.Log(null, Level.Warn, $"JFrog Connection was successfull!!", null);
```
Comment (adityanarayanp) [inline src/LCT.PackageIdentifier/BomCreator.cs:177]: for success shouldn't it be displayed as green color, and with Level.Notice ?

File: src/LCT.PackageIdentifier/ConanProcessor.cs [hunk 6]
```diff
@@ -364,8 +363,10 @@ private static string GetArtifactoryRepoName(List<AqlResult> aqlResultList, Comp
         {
             string jfrogcomponentPath = $"{component.Name}/{component.Version}";
 
-            string repoName = aqlResultList.Find(x => x.Path.Contains(
-                jfrogcomponentPath, StringComparison.OrdinalIgnoreCase))?.Repo ?? NotFoundInRepo;
+            var list = aqlResultList.FindAll(x => x.Path.Contains(
```
Comment (adityanarayanp) [inline src/LCT.PackageIdentifier/ConanProcessor.cs:366]: var list -> variable names should be more descriptive and not generic

File: src/LCT.PackageIdentifier/Program.cs [hunk 7]
```diff
@@ -82,9 +82,17 @@ static async Task Main(string[] args)
...
-            Logger.Logger.Log(null, Level.Notice, $"End of Package Identifier execution : {DateTime.Now}\n", null);
+            //Validating JFrog Settings
+            if (await bomCreator.CheckJFrogConnection())
+            {
+                await bomCreator.GenerateBom(appSettings, new BomHelper(), new FileOperations());
+                Logger.Logger.Log(null, Level.Notice, $"End of Package Identifier execution : {DateTime.Now}\n", null);
+            }
+            else
+            {
+                Logger.Logger.Log(null, Level.Alert, $"JFrog Connection was not successfull!! \n", null);
```
Comment (adityanarayanp) [inline src/LCT.PackageIdentifier/Program.cs:94]: isn't this already being logged by the method  CheckJFrogConnection(), why again ?

File: src/LCT.APICommunications/Interfaces/IJfrogAqlApiCommunication.cs (hunk 1 above)
Comment (sumanthkb44) [reply src/LCT.APICommunications/Interfaces/IJfrogAqlApiCommunication.cs:32]: Added

File: src/LCT.Facade/JfrogAqlApiCommunicationFacade.cs (hunk 3 above)
Comment (sumanthkb44) [reply src/LCT.Facade/JfrogAqlApiCommunicationFacade.cs:34]: Added

File: src/LCT.PackageIdentifier/BomCreator.cs (hunk 5 above)
Comment (sumanthkb44) [reply src/LCT.PackageIdentifier/BomCreator.cs]: Changed

File: src/LCT.APICommunications/JfrogAqlApiCommunication.cs (hunk 2 above)
Comment (sumanthkb44) [reply src/LCT.APICommunications/JfrogAqlApiCommunication.cs:85]: Added In Usage Document as optional.

**Review by adityanarayanp — APPROVED** [review]
approved
